core/arch/arm/plat-k3/drivers/mailbox.c

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Texas Instruments K3 Mailbox Driver
+ * Copyright (C) 2025 Texas Instruments Incorporated - https://www.ti.com/
+ */
+
+#include <io.h>
+#include <kernel/delay.h>
+#include <mm/core_memprot.h>
+#include <mm/core_mmu.h>
+#include <platform_config.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "ti_sci_protocol.h"
+#include "ti_sci_transport.h"
+
+struct ti_k3_mailbox {
+	void *tx_base;
+	void *rx_base;
+	void *tx_sram_va;
+	void *rx_sram_va;
+} mailbox_ctx;
+
+static TEE_Result ti_mailbox_poll_rx_status(void)
+{
+	uint32_t num_messages_pending = 0U;
+	uint32_t retry_count = 100U;
+	struct ti_k3_mailbox *mb;
+
+	mb = &mailbox_ctx;
+
+	while (num_messages_pending == 0U) {
+		num_messages_pending = io_read32((vaddr_t)mb->rx_base +
+						 TI_MAILBOX_MSG_STATUS);
+
+		if (retry_count-- == 0U) {
+			EMSG("Mailbox RX status polling timed out");
+			return TEE_ERROR_TIMEOUT;
+		}
+		mdelay(10);
+	}
+
+	return TEE_SUCCESS;
+}
+
+TEE_Result ti_sci_transport_send(const struct ti_sci_msg *msg)
+{
+	uint32_t num_bytes;
+	paddr_t phys_addr;
+	struct ti_k3_mailbox *mb;
+
+	mb = &mailbox_ctx;
```

Comment:
Why not combine these two? happening one after another so I think better to combine only. Also I think mailbox_ctx can be used directly itself.